x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/alert_summary/alert_summary_widget.test.tsx

```diff
@@ -74,13 +74,54 @@ describe('AlertSummaryWidget', () => {
     expect(alertSummaryWidget.queryByTestId(TITLE_DATA_TEST_SUBJ)).toBeTruthy();
   });
 
+  it('should render AlertSummaryWidget when there is only active alerts', async () => {
+    useLoadAlertSummaryMock.mockImplementation(() => ({
+      alertSummary: {
+        activeAlertCount: 1,
+        activeAlerts: [{ key: 1671408000000, doc_count: 1 }],
+        recoveredAlertCount: 0,
+      },
+      isLoading: false,
+    }));
+    const alertSummaryWidget = renderComponent();
+
+    expect(alertSummaryWidget.queryByTestId('alertSummaryWidgetCompact')).toBeTruthy();
+  });
+
+  it('should render AlertSummaryWidget compact version even when there is no active and recovered alerts', async () => {
+    useLoadAlertSummaryMock.mockImplementation(() => ({
+      alertSummary: {
+        activeAlertCount: 0,
+        activeAlerts: [],
+        recoveredAlertCount: 0,
+      },
+      isLoading: false,
+    }));
+    const alertSummaryWidget = renderComponent();
+
+    expect(alertSummaryWidget.queryByTestId('alertSummaryWidgetCompact')).toBeTruthy();
+  });
+
+  it('should not render AlertSummaryWidget full-size version when there is no active and recovered alerts', async () => {
```

Comment:
Do we have full size version in rule details page?

Reply:
No, we use a compact version on that page.

Reply:
I was wandering why do we need the test about full size version if the page doesn't use it. It seemed from the test description that if there are alerts present, full size version will be shown.

Reply:
Whether we show full-size or compact is dependent on providing `fullSize` prop, it is not decided automatically in the component.

Reply:
I think `should not render AlertSummaryWidget full-size version` would be clearer, as it is not dependent on whether there are alerts or not.

Reply:
But it depends on whether there are alerts; it is not rendered when there is no alert.

Reply:
Maybe the confusion is related to the fact that this test is not for the rule details page but for AlertSummaryWidget.

I have some other tests for rule details page but they are not included in this PR :)